File: src/lib/types/skill-score.ts
export interface SkillScore {
	id: string;
	program_id: string;
	domain: string;
	score: number;
	total_attempts: number;
	last_updated: string;
	created_at: string;
}

export interface SkillRadarData {
	domain: string;
	score: number;
	max_score: number;
}

File: src-tauri/src/db/models/skill_score.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct SkillScore {
    pub id: String,
    pub program_id: String,
    pub domain: String,
    pub score: i32,
    pub total_attempts: i32,
    pub last_updated: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SkillRadarData {
    pub domain: String,
    pub score: i32,
    pub max_score: i32,
}
